### PR TITLE
feat(FN-2088): add utilisation report state machine

### PR DESCRIPTION
**Pull request by toddym42**
## Introduction :pencil2:
Adds the outline of a utilisation report state machine. Event handlers to be completed as part of the Mongo -> SQL migration and as part of upcoming work to implement fee reconciliation.

Further details of the implementation can be found in the new [state-machines.md](https://github.com/UK-Export-Finance/dtfs2/pull/2759/files#diff-6211c403f460a5899181aa8ca9c48f7ab1ac4fb65b01d2e213d834a0aa2c61ca) doc.

### CODE REVIEW

File: dtfs-central-api/src/repositories/banks-repo.test.ts
Comment (toddym42) [inline dtfs-central-api/src/repositories/banks-repo.test.ts:1]: Moved `repositories` one level higher so they are no longer under `services`.

File: dtfs-central-api/tsconfig.json [hunk 1]
```diff
@@ -24,9 +24,9 @@
       // "useDefineForClassFields": true,                  /* Emit ECMAScript-standard-compliant class fields. */
 
       /* Modules */
-      "module": "CommonJS",                             /* Specify what module code is generated. */
+      "module": "node16",                             /* Specify what module code is generated. */
       "rootDir": "./",                                  /* Specify the root folder within your source files. */
-      "moduleResolution": "node",                       /* Specify how TypeScript looks up a file from a given module specifier. */
+      "moduleResolution": "node16",                       /* Specify how TypeScript looks up a file from a given module specifier. */
```
Comment (toddym42) [inline dtfs-central-api/tsconfig.json:29]: This change was required to allow us to recognise the multi-entry point functionality declared by
```
"exports": {
  ".": "./src/index.ts",
  "./sql-db-connection": "./src/sql-db-connection/index.ts"
},
```
in the `libs/common/package.json` so we can use the following import here in the `dtfs-central-api`
```
import { SqlDbDataSource } from '@ukef/dtfs2-common/sql-db-connection';
```

**Review by ChristianMcc — COMMENTED** [review]
Really nice work. Went through this once and it looked good - going to take some time tomorrow to go through the state machine/how it receives events in a bit more detail :)

File: dtfs-central-api/src/repositories/utilisation-reports-repo/utilisation-report-sql.repo.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import { UtilisationReportEntity } from '@ukef/dtfs2-common';
+import { ReportPeriod } from '../../types/utilisation-reports';
+import { NotFoundError } from '../../errors';
+
+export const UtilisationReportRepo = SqlDbDataSource.getRepository(UtilisationReportEntity).extend({
+  async findOneByBankIdAndReportPeriod(bankId: string, reportPeriod: ReportPeriod): Promise<UtilisationReportEntity | null> {
+    return await this.findOneBy({ bankId, reportPeriod });
+  },
+
+  async getOneById(id: number): Promise<UtilisationReportEntity> {
```
[the remark this answers is not in the document]
Comment (toddym42) [reply dtfs-central-api/src/repositories/utilisation-reports-repo/utilisation-report-sql.repo.ts:13]: I hadn't realised that was a thing! I guess in future if we wanted to centralise our error handling and return a response based on the error we'd probably want custom errors that also included a HTTP Status code, but we're not there yet so YAGNI! I'll remove this method and error

**Review by ChristianMcc — APPROVED** [review]
Looks good to me, let's get this merged to build on top of :)

**Pull request merged**